### PR TITLE
ONYX-10395: Bump ruby version to 3.0.2

### PR DESCRIPTION
**Pull request by semyon-estrin**
### Desired Outcome

Bumped ruby version from 2.5.8 to 3.0.2 in the following projects:
Phusion Ruby Builder
Ubuntu Ruby Builder
UBI Ruby Builder
Phusion Ruby FIPS
Ubuntu Ruby FIPS
UBI Ruby FIPS

### Implemented Changes

- Updated scripts and Dockerfile(s) to use 3.0 ruby version.
- Updated CHANGELOG and NOTICES to prepare for release.

### Connected Issue/Story

CyberArk internal issue link: [ONYX-10395](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-10395)
Ruby 3 Upgrade Epic link: [ONYX-9661](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-9661)

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [x] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [x] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+## [1.1.0] - 2021-11-11
```
Comment (sashaCher) [inline CHANGELOG.md:9]: There's no coupling between release and your change please keep the changelog in unreleased section

File: CHANGELOG.md [hunk 2]
```diff
@@ -70,6 +76,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Security
 - Bumped Ruby version from 2.5.1 to 2.5.8 to address [CVE-2020-10663](https://nvd.nist.gov/vuln/detail/CVE-2020-10663).
 
+[Unreleased]: https://github.com/cyberark/conjur-base-image/compare/v1.1.0...HEAD
```
Comment (sashaCher) [inline CHANGELOG.md:79]: The same ^^

File: VERSION [hunk 3]
```diff
@@ -1 +1 @@
-1.0.6
+1.1.0
```
Comment (sashaCher) [inline VERSION:1]: The same ^^

**Review by sashaCher — COMMENTED** [review]
Left comments, please remove version bump.
Have you checked the change with Conjur?

File: ubuntu-ruby-fips/build.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@ cd "$(dirname "$0")"
 
 UBUNTU_VERSION=20.04
 OPENSSL_BUILDER_TAG=1.0.2u-fips-2.0.16
-RUBY_BUILDER_TAG=2.5.8-fips
+RUBY_BUILDER_TAG=2.7.4-fips
```
Comment (sashaCher) [inline ubuntu-ruby-fips/build.sh:7]:
```suggestion
RUBY_BUILDER_TAG=3.0.2-fips
```

**Review by sashaCher — COMMENTED** [review]
Looks good
Please see a version remark
Please arrange commit history

**Review by ismarc — APPROVED** [review]
Looks good change wise.

Note that once merged, downstream projects will automatically pick it up, so if the changes aren't backwards compatible they will fail until their changes are merged.  This should be taken into account when decided to merge (i.e., are all the downstream projects ready to merge as well).

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - Bump bundler from 2.1.4 to 2.2.30 as a prerequisite to ruby 3.0.2.
   [cyberark/conjur-base-image#72](https://github.com/cyberark/conjur-base-image/pull/72)
+- Upgrade Ruby version from 2.5.8(EOL) to 3.0.2.
+  [cyberark/conjur-base-image#71](https://github.com/cyberark/conjur-base-image/pull/71)
```
[the remark this answers is not in the document]
Comment (semyon-estrin) [reply CHANGELOG.md]: Done, moved the changes to 2.0.0 section.
To isolate ruby 3 change when releasing 2.0.0, 1.0.7 will be released prior to it.
Created another PR, according to automated release process guidelines:
https://github.com/cyberark/conjur-base-image/pull/78

**Comment by sashaCher** [comment]
Please arrange commit history

**Review by sashaCher — APPROVED** [review]
LGTM

**Review by ismarc — APPROVED** [review]

**Pull request merged**